api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2294,6 +2297,182 @@ public void copyObject(String bucketName, String objectName, String destBucketNa
+   * </pre>
+   *
+   * @param destinationBucketName
+   *          Destination Bucket to be created upon compose.
+   * @param destintionObjectName
+   *          Destination Object to be created upon compose.
+   * @param sse
+   *          Server Side Encryption.
+   * @param userMetaData
+   *          User Meta data.
+   * @param listSourceObjects
+   *          List of Source Objects used to compose Object
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InsufficientDataException  upon getting EOFException while reading given
+   *           InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidArgumentException    upon invalid value is passed to a method.
+   */
+
+
+  public void  composeObject(String destinationBucketName, String destintionObjectName, ServerSideEncryption sse,
+                             Map<String,String> userMetaData, List<CopySource> listSourceObjects)
+          throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+          InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+          InternalException, InvalidArgumentException {
+
+    if (listSourceObjects.size() < 1 || listSourceObjects.size() > MAX_MULTIPART_COUNT ) {
+      throw new InvalidArgumentException("There must be as least one and up to 10000 source objects.");
+    }
+    int i = 0;
+    long totalSize = 0;
+    for (CopySource sourceObject : listSourceObjects) {
+      i++;
+      ObjectStat os = statObject(sourceObject.bucketName(), sourceObject.objectName());
+      // Error out if client side encryption is used in this source object when
+      // more than one source objects are given.
+      if ((listSourceObjects.size() > 1 )  && sourceObject.headers().get("x-amz-meta-x-amz-key") !=  null) {
+        throw new InvalidArgumentException("Client side encryption is used in source object"
+                + sourceObject.bucketName() + sourceObject.objectName());
+      }
+
+      // Check if a segment is specified is within object bounds
+      if ( sourceObject.length() > os.length()) {
+        throw new InvalidArgumentException("Source specified length " + sourceObject.length() + " is more than  "
+                +  "size of Object " + os.length());
+      }
+
+      if ( os.length() < MIN_MULTIPART_SIZE && i < listSourceObjects.size() - 1 ) {
```

Comment:
`i` is incremented before this, hence this check doesn't work.

Reply:
done.